packers/jws.go

```diff
@@ -0,0 +1,384 @@
+	"github.com/iden3/iden3comm"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jws"
+	"github.com/mr-tron/base58"
+	"github.com/pkg/errors"
+)
+
+type verificationType string
+
+// List of supported verification types
+const (
+	JSONWebKey2020                    verificationType = "JsonWebKey2020"
+	EcdsaSecp256k1VerificationKey2019 verificationType = "EcdsaSecp256k1VerificationKey2019"
+	EcdsaSecp256k1RecoveryMethod2020  verificationType = "EcdsaSecp256k1RecoveryMethod2020"
+	EddsaBN256VerificaonKey           verificationType = "EddsaBN256VerificaonKey"
+)
+
+var supportedAlgorithms = map[jwa.SignatureAlgorithm]map[verificationType]struct{}{
+	jwa.ES256K: {
+		JSONWebKey2020:                    {},
+		EcdsaSecp256k1VerificationKey2019: {},
+		EcdsaSecp256k1RecoveryMethod2020:  {},
+	},
+	jwa.ES256: {
+		JSONWebKey2020:                    {},
+		EcdsaSecp256k1VerificationKey2019: {},
+		EcdsaSecp256k1RecoveryMethod2020:  {},
+	},
+	BJJAlg: {
+		EddsaBN256VerificaonKey: {},
+		// "JsonWebKey2020":                    {}, for future use
+	},
+}
+
+// MediaTypeSignedMessage is media type for jws
+const MediaTypeSignedMessage iden3comm.MediaType = "application/iden3-signed-json"
+
+// DIDResolverHandlerFunc resolves did
+type DIDResolverHandlerFunc func(did string) (*verifiable.DIDDocument, error)
+
+// Resolve function is responsible to call provided handler for resolve did document
+func (f DIDResolverHandlerFunc) Resolve(did string) (*verifiable.DIDDocument, error) {
+	return f(did)
+}
+
+// SignerResolverHandlerFunc resolves signer
+type SignerResolverHandlerFunc func(kid string) (crypto.Signer, error)
+
+// Resolve function return signer by kid
+func (f SignerResolverHandlerFunc) Resolve(kid string) (crypto.Signer, error) {
+	return f(kid)
+}
+
+// JWSPacker is packer that use jws
+type JWSPacker struct {
+	didResolverHandler        DIDResolverHandlerFunc
+	signerResolverHandlerFunc SignerResolverHandlerFunc
+}
+
+// SigningParams packer parameters for jws generation
+type SigningParams struct {
+	SenderDID string
+	Alg       jwa.SignatureAlgorithm
+	KID       string
+	DIDDoc    *verifiable.DIDDocument
+	iden3comm.PackerParams
+}
+
+// NewSigningParams defines the signing parameters for jws generation
+func NewSigningParams() SigningParams {
+	return SigningParams{}
+}
+
+// Verify checks if signing params are valid
+func (s *SigningParams) Verify() error {
+	if s.Alg == "" {
+		return errors.New("alg is required for signing params")
+	}
+	if s.SenderDID == "" {
+		return errors.New("sender did is required for signing params")
+	}
+	return nil
+}
+
+// NewJWSPacker creates new jws packer instance
+func NewJWSPacker(
+	didResolverHandler DIDResolverHandlerFunc,
+	opaqueSignerResolverHandlerFunc SignerResolverHandlerFunc,
+	// opaqueVerifierResolverHandlerFunc OpaqueVerifierResolverHandlerFunc,
+) *JWSPacker {
+	return &JWSPacker{
+		didResolverHandler,
+		opaqueSignerResolverHandlerFunc,
```

Comment:
do we still need opaqueSignerResolverHandlerFunc

Reply:
fixed